packages/@okta/vuepress-site/guides/sign-into-native-ui/handle-authentication-responses/iOS/handle-responses.md

```diff
@@ -0,0 +1,107 @@
+
+// Unlock account
+public class func unlockAccount(with url: URL,
+                                username: String,
+                                factorType: OktaRecoveryFactors,
+                                onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                onError: @escaping (_ error: OktaError) -> Void)
+
+// Forgot password
+public class func recoverPassword(with url: URL,
+                                  username: String,
+                                  factorType: OktaRecoveryFactors,
+                                  onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                  onError: @escaping (_ error: OktaError) -> Void)
+
+// Restore authentication
+public class func fetchStatus(with stateToken: String,
+                              using url: URL,
+                              onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                              onError: @escaping (_ error: OktaError) -> Void)
+```
+
+Please note that the closure `onStatusChange` returns `OktaAuthStatus` instance as a parameter. An instance of the `OktaAuthStatus` class represents the current status that is returned by the server. It's the developer's responsibilty to handle current status in order to proceed with the initiated flow. Check the status type by calling `status.statusType` and downcast to a concrete status class. Here is an example handler function:
+
+```swift
+func handleStatus(status: OktaAuthStatus) {
+
+    switch status.statusType {
+        case .success:
+            let successState: OktaAuthStatusSuccess = status as! OktaAuthStatusSuccess
+            handleSuccessStatus(successStatus: successStatus)
+
+        case .passwordWarning:
+            let warningPasswordStatus: OktaAuthStatusPasswordWarning = status as! OktaAuthStatusPasswordWarning
+            handlePasswordWarning(passwordWarningStatus: warningPasswordStatus)
+
+        case .passwordExpired:
+            let expiredPasswordStatus: OktaAuthStatusPasswordExpired = status as! OktaAuthStatusPasswordExpired
+            handleChangePassword(passwordExpiredStatus: expiredPasswordStatus)
+
+        case .MFAEnroll,
+             .MFAEnrollActivate,
+             .MFARequired,
+             .MFAChallenge.
+             .recovery,
+             .recoveryChallenge,
+             .passwordReset,
+             .lockedOut:
+             let alert = UIAlertController(title: "Error", message: "Not implemented", preferredStyle: .alert)
+             alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+             present(alert, animated: true, completion: nil)
+    }
+}
+```
+
+### Success status handling
+
+Downcast the received status to OktaAuthStatusSuccess class and get access to `sessionToken` property. You now can exchange session token to access token via `OktaOidc` library.
```

Comment:
We need to show this code: how does the developer finish the authentication by passing the sessionToken back to the OIDC SDK?